Fix recent changes in the case of a stack that grows up (on PARISC):

js/src/gc/RootMarking.cpp: In function 'void MarkConservativeStackRoots(JSTracer*, bool)':
js/src/gc/RootMarking.cpp:317:20: error: invalid conversion from 'uintptr_t {aka unsigned int}' to 'uintptr_t* {aka unsigned int*}' [-fpermissive]

This case didn't see the changes that the alternative got
https://bugs.gentoo.org/show_bug.cgi?id=414297

    Jeroen Roovers <jer@gentoo.org>
--

--- a/js/src/gc/RootMarking.cpp
+++ b/js/src/gc/RootMarking.cpp
@@ -314,7 +314,7 @@
 
     uintptr_t *stackMin, *stackEnd;
 #if JS_STACK_GROWTH_DIRECTION > 0
-    stackMin = rt->nativeStackBase;
+    stackMin = reinterpret_cast<uintptr_t *>(rt->nativeStackBase);
     stackEnd = cgcd->nativeStackTop;
 #else
     stackMin = cgcd->nativeStackTop + 1;
--- a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_pa32.cpp
+++ b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_pa32.cpp
@@ -124,7 +124,7 @@
   return result;
 }
 
-extern "C" int SharedStub(int);
+extern "C" nsresult SharedStub(int);
 
 #define STUB_ENTRY(n)       \
 nsresult nsXPTCStubBase::Stub##n()  \
